src/Agent.Worker/Build/TrackingManager.cs

```diff
@@ -145,29 +143,7 @@ private void WriteToFile(string file, object value)
             Trace.Entering();
             Trace.Verbose($"Writing config to file: {file}");
 
-            // Create the directory if it does not exist.
-            Directory.CreateDirectory(Path.GetDirectoryName(file));
-
-            // Serialize the object.
-            JsonSerializerSettings settings = new JsonSerializerSettings();
-            settings.ContractResolver = new CamelCasePropertyNamesContractResolver();
-            settings.Converters.Add(new StringEnumConverter() { CamelCaseText = true });
-            string contents = JsonConvert.SerializeObject(
-                value: value,
-                formatting: Formatting.Indented,
-                settings: settings);
-
-            // Lock write access on the file so the garbage collector service
-            // can also lock on write access to safely handle the file.
-            using (FileStream stream = new FileStream(file, FileMode.Create, FileAccess.Write, FileShare.Read))
-            {
-                using (StreamWriter writer = new StreamWriter(stream))
-                {
-                    writer.Write(contents);
-                    writer.Flush();
-                    stream.Flush();
-                }
-            }
+            FileSystemHelper.WriteJsonSerializeToFile(file, value);
```

Comment:
This looks a lot like IOUtil.SaveObject. If you're moving into a helper, can you reconcile this with that existing helper rather than add a new one (to a separate new file) which seems to be very similar?